source/renderer/app/components/wallet/mnemonic-input/MnemonicAutocompleteContainer.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {
+  ClipboardEventHandler,
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { startsWith } from 'lodash/fp';
+import { GlobalListeners } from 'react-polymorph/lib/components/HOC/GlobalListeners';
+import { MnemonicAutocompleteLayout } from './MnemonicAutocompleteLayout';
+
+interface MnemonicAutocompleteContainerProps {
+  onChange: (value: string) => void;
+  onConfirmSelection: () => void;
+  onPaste: ClipboardEventHandler<HTMLInputElement>;
+  ordinalNumber: number;
+  value: string;
+  options: string[];
+  maxVisibleOptions: number;
+  disabled: boolean;
+  error: boolean;
+  reset: boolean;
+  noResultsMessage: string;
+  inputRef: RefObject<HTMLInputElement>;
+}
+
+const MnemonicAutocompleteContainer = ({
+  onChange,
+  onConfirmSelection,
+  onPaste,
+  reset,
+  ordinalNumber,
+  value,
+  options,
+  maxVisibleOptions,
+  disabled,
+  error,
+  noResultsMessage,
+  inputRef,
+}: MnemonicAutocompleteContainerProps) => {
+  const initialState = useMemo(
+    () => ({
+      inputValue: value,
+      selectedOption: '',
+      filteredOptions: options.slice(0, maxVisibleOptions - 1),
+      isOpen: false,
+      mouseIsOverOptions: false,
+      blurred: false,
+    }),
+    []
+  );
+  const [state, setState] = useState(initialState);
+  const rootRef = useRef();
+  const optionsRef = useRef();
+  const suggestionsRef = useRef();
+
+  const toggleOpen = useCallback(() => {
```

Comment:
Why are we using `useCallback` here instead of toggling the state in an event handler directly? We don't need to memorize anything (same for all other cases in this function)